extensions/amp-analytics/0.1/session-manager.js

```diff
@@ -76,41 +76,33 @@ export class SessionManager {
    * Get the value from the session per the vendor.
    * @param {string|undefined} type
    * @param {SESSION_VALUES} value
-   * @param {boolean=} opt_persist
    * @return {!Promise<number|undefined>}
    */
-  getSessionValue(type, value, opt_persist) {
-    return this.get(type).then((session) => {
-      if (value === SESSION_VALUES.EVENT_TIMESTAMP) {
-        return this.getSessionEventTimestamp_(type, session, opt_persist);
-      }
-      return session?.[value];
-    });
+  getSessionValue(type, value) {
+    return this.get(type).then((session) => session?.[value]);
   }
 
   /**
-   * Get the value from the session per the vendor.
+   * Update EventTimestamp for this session async.
+   * Passes in a callback to set the EventTimestamp,
+   * to be used when the session it retrieved or created.
    * @param {string} type
-   * @param {SessionInfoDef} session
-   * @param {boolean} persist
-   * @return {number|undefined}
+   * @return {!Promise}
    */
-  getSessionEventTimestamp_(type, session, persist) {
-    const lastTimestamp = session[SESSION_VALUES.EVENT_TIMESTAMP];
-    if (persist) {
-      this.sessions_[type] = this.updateSession_(session, true);
-      this.setSession_(type, session);
-    }
-    return lastTimestamp;
+  updateEventTimestamp(type) {
+    return this.get(type, (session) => {
+      session[SESSION_VALUES.EVENT_TIMESTAMP] = Date.now();
+    });
   }
 
   /**
    * Get the session for the vendor, checking if it exists or
    * creating it if necessary.
    * @param {string|undefined} type
+   * @param {?Function} opt_processing
```

Comment:
```suggestion
   * @param {function=} opt_processing
```

Reply:
This is currently an incorrect type annotation. `{?Type}` --> `{null|Type}`, but expects a non-`undefined` value. This should be annotated as an optional property instead. This would fail typechecking if/when enabled